balena/models/device.py

```diff
@@ -1197,3 +1202,95 @@ def get_supervisor_state(self, uuid):
             '/supervisor/v1/device', 'POST', data=data,
             endpoint=self.settings.get('api_endpoint')
         )
+
+    def __check_os_update_target(self, device_info, target_os_version):
+        """
+        """
+
+        if 'uuid' not in device_info and not device_info['uuid']:
```

Comment:
I think that this needs to be an `or`
```suggestion
        if 'uuid' not in device_info or not device_info['uuid']:
```